### PR TITLE
Introduce CMS Controller Manager

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Created CMS Controller Manager

**Related issue(s)**

Issue: #2657
Chart: #3060

### CODE REVIEW

File: components/cms-controller-manager/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+```bash
+IMG={image_name}:{image_tag} make docker-build
+```
+
+The variables are:
+
+* `{image_name}` which is the name of the output image. Use `assetstore-controller-manager` for the image name.
+* `{image_tag}` which is the tag of the output image. Use `latest` for the tag name.
+
+### Environmental Variables
```
Comment (kazydek) [inline components/cms-controller-manager/README.md]:
```suggestion
### Environment Variables
```

File: components/cms-controller-manager/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# CMS Controller Manager
```
Comment (kazydek) [inline components/cms-controller-manager/README.md]:
```suggestion
# Content Management System (CMS) Controller Manager
```

File: components/cms-controller-manager/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Overview
+
+CMS is a Kubernetes-native solution for storing managing content. It is build on top of Asset Store and it consists of the DocsTopic Controller and the ClusterDocsTopic Controller.
```
Comment (kazydek) [inline components/cms-controller-manager/README.md]:
```suggestion
CMS is a Kubernetes-native solution for managing content. It is built on top of the Asset Store and consists of the DocsTopic Controller and the ClusterDocsTopic Controller.
```

File: components/cms-controller-manager/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+* `{image_tag}` which is the tag of the output image. Use `latest` for the tag name.
+
+### Environmental Variables
+
+Use the following environment variables to configure the application:
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| **APP_CLUSTER_DOCS_TOPIC_RELIST_INTERVALL** | No | `5m` | The period of time after which the controller refreshes the status of a ClusterDocsTopic CR |
+| **APP_CLUSTER_BUCKET_REGION** | No |  | Specifies the location of the region under which the controller creates the cluster bucket. If the field is empty, the controller creates the cluster bucket under the default location. |
```
Comment (kazydek) [inline components/cms-controller-manager/README.md]:
```suggestion
| **APP_CLUSTER_BUCKET_REGION** | No |  | Specifies the location of the region in which the controller creates a ClusterBucket CR. If the field is empty, the controller creates the bucket under the default location. |
```

File: components/cms-controller-manager/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+| **APP_DOCS_TOPIC_RELIST_INTERVALL** | No | `5m` | The period of time after which the controller refreshes the status of a DocsTopic CR |
+| **APP_BUCKET_REGION** | No |  | Specifies the location of the region under which the controller creates the bucket. If the field is empty, the controller creates the bucket under the default location. |
```
Comment (kazydek) [inline components/cms-controller-manager/README.md]:
```suggestion
| **APP_BUCKET_REGION** | No |  | Specifies the location of the region in which the controller creates a Bucket CR. If the field is empty, the controller creates the bucket under the default location. |
```

File: components/cms-controller-manager/README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+* `{image_name}` which is the name of the output image. Use `assetstore-controller-manager` for the image name.
+* `{image_tag}` which is the tag of the output image. Use `latest` for the tag name.
+
+### Environmental Variables
+
+Use the following environment variables to configure the application:
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| **APP_CLUSTER_DOCS_TOPIC_RELIST_INTERVALL** | No | `5m` | The period of time after which the controller refreshes the status of a ClusterDocsTopic CR |
```
Comment (kazydek) [inline components/cms-controller-manager/README.md]:
```suggestion
| **APP_CLUSTER_DOCS_TOPIC_RELIST_INTERVALL** | No | `5m` | The period of time after which the controller refreshes the status of a ClusterDocsTopic CR. |
```

File: components/cms-controller-manager/pkg/apis/cms/v1alpha1/docstopic_common.go [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+type DocsTopicMode string
+
+const (
+	DocsTopicSingle  DocsTopicMode = "single"
+	DocsTopicPackage DocsTopicMode = "package"
+	DocsTopicIndex   DocsTopicMode = "index"
+)
+
+type Source struct {
+	Url string `json:"url"`
```
Comment (tgorgol) [inline components/cms-controller-manager/pkg/apis/cms/v1alpha1/docstopic_common.go]: Please rename `Url` to `URL`

Comment (michal-hudy) [reply components/cms-controller-manager/pkg/apis/cms/v1alpha1/docstopic_common.go]: Ok, it needs to be adjusted also in Asset.

**Pull request merged**